Let's start a discussion

burtlo opened the pull request:
A few weeks ago I started look at ways I could improve upon Showoff by adding:
- Github Flavored Markdown
  
  > Syntax Highlighting through code fences instead of the strange
  > additional construct of using `@@@ LANGUAGE` within the code samples.
  > Essentially your markdown can now adhere to more accepted markdown
  > standards.
- Static HTML and PDF
  
  > All javascript, stylesheets, and images are embedded into the static HTML
  > and PDF output. This allows the static HTML to better represent the 
  > presentation. This includes any custom javascript, css, or images defined
  > within the presentation. This also embeds locally defined images within the
  > CSS.

I think after that I went off the deep end trying to clean up and refactor the monolith sinatra class that contained all the parsing and rendering logic. At some point I questioned the usefulness of a `showoff.json` file and replaced with a `showoff` file which contained a DSL that allowed you to define sections and slides.

The README describes in far greater detail this implementation.
### Final Note

> This pull request is unreasonable in size, scope, and change. It is likely near impossible to visibly inspect it to gain a sense of it. I am absolutely not sure if there is a home for this abomination. However, I felt like freeing this Frankenstein...

goncalossilva commented:
Wow.

I really think we should join efforts into improving showoff (vs forking it). Thanks for all this!

First of all, I couldn't agree more regarding GitHub's flavored markdown. It's the way to go, imo. I also like the stuff you did with static page & PDF generation. And finally, great work shrinking `showoff.rb`! 

Having said this, I see a few issues with this pull request:
- I love the syntax of the new config file (I'm a ruby guy, after all), but I strongly believe the JSON config file should not be deprecated (for now). Besides the obvious advantage of not breaking previous presentations, I'd say it's easier on non-programmers... but I could be wrong.
- The README file cannot be merged as is, since it compares the current showoff with these changes.
- I believe you missed some recent additions to showoff, like presentation templates, markdown engine configurability, the pause screen and presenter view <—> presentation view 2-way sync. Am I wrong?

Let's discuss this a bit more!

alexch commented:
FYI I'm kinda doing some of this already with deck.rb, which is based on a cleaner JS presentation layer and an arguably cleaner and better tested Ruby tool. http://github.com/alexch/deck.rb

(I'm not using full GFM but I'd love to be, and I'm inlining a lot of the JS and CSS too. Also it should be drop-in-compatible with showoff since I support showoff.json files so I'd love for more people to just give it a try and tell me what breaks.)

burtlo commented:
@goncalossilva this is to start a conversation, so don't feel too bad about the work I have done on the fork coordinating with your work. Thanks for taking a look at it and your feedback.

### Supporting `showoff.json`

I am fairly certain that I could build an adapter to parse the previous `showoff.json` format.

I think the new format, though Ruby, is a little easier to read and possibly to understand for non-developers. There are overall less symbols _{ , }, [, ], :_ in the new format.

### README

That of course would change if you are seriously considering this monstrous body of work.

## Missing Features

### Markdown Engine Configurability

I likely have broken the markdown engine configurability because I have never navigated this minefield before and when it came time to get Github Flavored Markdown (GFM), I simply opted for a solution that I found that worked.  Namely using [Redcarpet with Pygments.rb](https://github.com/burtlo/showoff/blob/master/lib/showoff/renderers/html_with_pygments.rb). Perhaps there is a way to restore that functionality

### Presenter View and Presentation View

That functionality still exists and is working but is indeed missing from the documentation I composed.

### Presentation Templates

I'm not sure what you mean by presentation templates? Are those the classes that can be applied to slides? The ability to request `hostname:9090/onepage` or `hostname:9090/pdf`?

burtlo commented:
Alright, both the server and the static generation will look for a `showoff` then the original `showoff.json`.

Also I took a moment to make the server and the static generation use the full filepath provided and not simply a directory.

So you could do something like:

``` bash
$ bin/showoff server examples/showon
$ bin/showoff server examples/custom.json
```

Which will add those files to the list of files to search for when creating the presentation.

lynnvfrank committed:
Updated column examples

lynnvfrank committed:
Added columns to README

lynnvfrank committed:
Font size scales with the number of columns
Images are centered in the column
Code samples do not have all those margins

lynnvfrank committed:
Merge branch 'columns-css'

lynnvfrank committed:
Default the showoff file

lynnvfrank committed:
Heroku requires a specific version of pygments

lynnvfrank committed:
Additional req on rubypython

lynnvfrank committed:
A different attempt of pygments

burtlo commented:
@goncalossilva let's talk about what you would love to see in a series of smaller pull requests:
- Server refactor into parsers and renderers
- Onepage and PDF inlined assets
- Showoff DSL support with backwards compatibility to still use the existing JSON format
- Github Flavored Markdown

> I am running into some problems pushing this functionality to Heroku because of some python issues. I'll see what I can do to make this work.
- Multi-column layout

> This was something that I was working on in the last few commits.
- Updated Example Presentation

> I wanted to update the example presentation so that it actually show-cased Showoff and the majority of it's features.
- RSpec test suite

> How do you feel about having an RSpec test suite backed with Guard?
- Reduced `showoff` command 

> I removed a large set of the showoff commands so that showoff could focus ensuring high quality of core commands. I also made the showoff command work when simply given a directory or filepath.

lynnvfrank committed:
A different version attempt

lynnvfrank committed:
Setting explicit Rubypython version

lynnvfrank committed:
Only find css/js in root dir

lynnvfrank committed:
Removing python version and trying it per app

lynnvfrank committed:
Merge branch 'heroku-pygments'

lynnvfrank committed:
Heroku currently requires python 2.6

goncalossilva commented:
@burtlo In parts:

**Markdown Engine Configurability**
I'll try to take a deeper look into this in the near future.

**Presentation Templates**
I'm talking about this: https://github.com/schacon/showoff#templates

**RSpec test suite**
Any test suite is better than no test suite :p

The rest seems fine. Your PR breakdown looks good to me. After ironing out the questions about the MD engines and the templates, I think we're good to go.

I think this is the best approach:
- Iron out the remaining questions
- Release a new 0.7.X version of showoff
- Merge most of this in smaller PR, if possible
- Iron out some rough edges, make a pre release
- Release 0.8

These are big changes. I'm strongly in favor of most of them. Let's see if @schacon can weigh in on this!

burtlo commented:
Oh templates! I must have forked before templates. I'll work on how that would integrate with the work I've done.

lynnvfrank committed:
WIP custom templating

goncalossilva commented:
There were a few commits after that one :|

Everything after 580c02595bbc422844d0769e95214f6f3411ba4c (https://github.com/schacon/showoff/commits/master?page=2).

burtlo commented:
Looking over the current template implementation, I'm not sure why the templates aren't simply using `erb`. Would you be opposed to using ERB?

``` erb
<div style='background-color: red;' id='<%= id %>' class='slide <%= slide_classes %>' data-transition='<%= transition %>'>
  <div class='content <%= content_classes %>' ref='<%= reference %>'>
    <%= content_as_html %>
  </div>
</div>
```

lynnvfrank committed:
ERB Templating support with defaulting

Within the showoff file, a template name can be associated with a
template file. That name can be used within the slides to generate
a slide with that template file.

goncalossilva commented:
Not at all. Did you check out the rest of the commits?

burtlo commented:
Yes, I'm looking them over.

lynnvfrank committed:
JSON file support templates

lynnvfrank committed:
Merge branch 'templates'

lynnvfrank committed:
Pause message feature.

Presentations are able to define a pause message for the pause
screen. Unlike most features which is configurable per section,
this one is only valid on the top level showoff file

lynnvfrank committed:
Merge branch 'pause-screen'

burtlo commented:
I had a question about the `full-page` slide class.
- full-page - allows an image to take up the whole slide

I took it to mean, from the description, that it should take all images on the slide and center and enlarge them:

![full-screen-in-my-branch](https://img.skitch.com/20120529-8dp83112ekt94431msgbq4isi9.png)

In the current implementation it looks like the following:

![full-screen-original](https://img.skitch.com/20120529-tgqx8uqja1mahxqbwnfecjcnrk.png)

However, full-page does definitely sound like it should break out of the slide mold and show it full-width.

goncalossilva commented:
Hum. Tough choice. In my opinion "full-page" doesn't mean "enlargement", but this is just my opinion. What about having them both with different names?

PS. Off the top of my head, I just remembered Android dev. When using images, there are various types of scale types (http://developer.android.com/reference/android/widget/ImageView.ScaleType.html)... we're basically discussing if "full-page" means CENTER_INSIDE or CENTER.

lynnvfrank committed:
Support single quotation mark for CSS images and HTML image tags.

lynnvfrank committed:
Add support external image sources

lynnvfrank committed:
The page should scroll on the single page view.

lynnvfrank committed:
FIX Presenter display layout hosed on Firefox 9 and 10

lynnvfrank committed:
presenter view: give the spawned slides window the ability to move our location

burtlo commented:
@goncalossilva I'll return it to the latter option, as the enlargement wasn't great, just what I thought of initially when I saw the text.

I think I have included most of the fixes and pull requests from the past three months; except for the multiple markdown support. I will take a look at that latter this evening.

lynnvfrank committed:
FIX reset preso width after going full-page

lynnvfrank committed:
SPEC fixes for metadata and integration test

lynnvfrank committed:
SPEC fixes for new template lookup

lynnvfrank committed:
Removed comment from the version

lynnvfrank committed:
Moved all utility to individual cmd files

lynnvfrank committed:
Refactored further into a new commands class

lynnvfrank committed:
Further reduction of utilities methods

lynnvfrank committed:
DRYed up the command registration and retrieval

lynnvfrank committed:
Consolidated template rendering methods to a module

lynnvfrank committed:
DRY up HTML generation for static html/pdf

lynnvfrank committed:
Removed the unused name property

lynnvfrank committed:
Docs for the new commands

lynnvfrank committed:
Merge branch 'command-refactoring'

lynnvfrank committed:
FIX sequence count is updated for entire presentation

lynnvfrank committed:
WIP Presentation JS object

Some presentation methods have been ported over into a class in the
index.erb and removed from the showoff.js

lynnvfrank committed:
Metadata Parsers

> Instead of relying on an ever growing if-elsif-else block, the metdata
  parsers is simply a list of classes control the behavior of what data
  is matched and how it is converted to the metadata hash (which is
  then provided to the metadata hash)

lynnvfrank committed:
Merge branch 'metadata-parsers'

lynnvfrank committed:
Added tests for each metadata parser

lynnvfrank committed:
Spine - JS Class Structure

Presentation is now composed of various sub-components:
* NavigationMenu
* DebugView
* SpeakerNotes
* PauseScreen
* HelpMenu
* Footer

Using an pub/sub arch to allow for a less coupled design

Slides are now managed by a Slides and returns a Slide object which
manages the increments, transition, and other information.

Extracted a location watcher which manages watching the
window.location and also watches the changes in slides

burtlo commented:
To further clean up the project, I have started to tackle the javascript. I have been trying to create a more modular system with classes that employ **pub/sub**.

https://github.com/burtlo/showoff/blob/js-refactor/lib/public/js/showoff.js

I am not done with it yet, but getting pretty close.

lynnvfrank committed:
JS - Preshow

Created a preshow and have it working in a rudimentary fashion

lynnvfrank committed:
FIX: Assets paths starting in presenting directory

  Javascript and CSS were failing to load when starting the presentation
  from within the folder. This is because of using gsub which replaced
  all instances of . instead of replacing only the first instance.

lynnvfrank committed:
Initial implementation of custom resources

  DSL supports a resource paths which will be stored and then added to
  the presentation for the CSS and JS files.

lynnvfrank committed:
Additional documentation for Metadata classes

lynnvfrank committed:
Additional documentation for the DSL and JSON parsers

lynnvfrank committed:
Server documentation and use of helpers

lynnvfrank committed:
Removed redundant #build call

lynnvfrank committed:
Merge branch 'shared-resources'

Conflicts:
	lib/showoff/parsers/dsl.rb
	lib/showoff/server.rb

lynnvfrank committed:
Custom CSS/JS files

  * Helpers will merge the filepaths better
  - FIX the request for custom resources by allowing js and css to
    be returned from the custom resources
  * Refactored to use a block to reduce the redundancy of the code
    between the js and css

lynnvfrank committed:
FIX: More Specific Gem requirements (parslet & gli)

This should address an issue for attempting to run showoff with a recent
bundle call. For some reason the parslet version that was being
installed was 0.9.0 and the gli version was too old.

lynnvfrank committed:
Cleanup of JS that I do not want or I believe is unused

lynnvfrank committed:
Using an unminified JS

lynnvfrank committed:
Unminified coffeescript and upgrade to 1.4.0

lynnvfrank committed:
Removed the swipe events that were commented out

lynnvfrank committed:
Keyboard.js

Using an abstraction library to manage the big behemoth of keyboard
input

lynnvfrank committed:
Return to the old coffeescript

lynnvfrank committed:
Removed a console.log

lynnvfrank committed:
Added visor to enter commands; just responds with the command

lynnvfrank committed:
Keyboard JS added toggle support

lynnvfrank committed:
Visor show disables other keys

lynnvfrank committed:
Initial command mapping

lynnvfrank committed:
Fuzzy matching on commands

n, ne, nex will match next
p, pr, pre, prev, previ, previo, previou will match previous

lynnvfrank committed:
Visor size and letter spacing; removed welcome message

lynnvfrank committed:
Moved code execution with other commands

lynnvfrank committed:
Working refactor of the code execution

lynnvfrank committed:
Minor refactor to move the code execution into its own file

lynnvfrank committed:
Removed presenter mode until it can be fixed

lynnvfrank committed:
Merge branch 'js-refactor'

lynnvfrank committed:
Renamed all files to Parade

burtlo commented:
I took a break and have now returned to working on these changes. I believe I am going to fork this into a new gem.

alexch commented:
I agree that showoff's JS needs work, but instead, I've been using deck.js and it's great; it's fully unit tested with Jasmine and with my deck.rb wrapper gem, it's mostly backwards-compatible with showoff (e.g. understands `showoff.json` files).

https://github.com/alexch/deck.rb

burtlo commented:
@alexch I think deck.rb is still missing some fundamental things but I would love to continue this conversation further.

lynnvfrank committed:
HTML fragment not an XML fragment

lynnvfrank committed:
HTML fragment not XML fragment

alexch commented:
@burtlo So how about you make a "deck.rb is still missing some fundamental things" issue at http://github.com/alexch/deck.rb/issues and we can continue this conversation further :-)

lynnvfrank committed:
FIX for showing 'help'; removed GLI cruft

lynnvfrank committed:
Renamed of template to parade

lynnvfrank committed:
FIX for command-prompt bump when first starting

The prompt is immediately made visible, while the input follows after in
the typewrtier fashion.

lynnvfrank committed:
Removed unusued parser that should not have been there

lynnvfrank committed:
Command Line Renderer

Added command-line styles to mirror the bash
Changed the time the command is rendered to be based on the length

lynnvfrank committed:
Fixed pause message url

lynnvfrank committed:
FIX to example - navigation

lynnvfrank committed:
Changed to use gem 'debugger'

lynnvfrank committed:
Changed around slide classes and renderer assignment

* Post  Renderers are now registered in their own object
* Slides now receive the same css classes as the content
* Column renderer is fixed to work with this change
* Added a default.css which will eventually be the default styes
  for parade

lynnvfrank committed:
Slides to use SlidesPreRenderers that are registered

lynnvfrank committed:
Slides by default will no longer be centered

lynnvfrank committed:
Updated example to now center slides

lynnvfrank committed:
Updated column examples and added a single column with updated css

lynnvfrank committed:
Added a slide to list all the transitions

lynnvfrank committed:
Removed 'ref' from the slide

Moving more content to their own individual attributes and using those
to correctly generate the table of contents.

lynnvfrank committed:
killed some whitespaces

lynnvfrank committed:
Substring the title when it exceeds 15 char

lynnvfrank committed:
Initial working version of multiple keyboard states

lynnvfrank committed:
Menu being navigated by keybound notifications - hacky

lynnvfrank committed:
Navigation Menu keyboard event handlers now working with navigation keys

lynnvfrank committed:
Goto by slide title name and also 'start' and 'end'

lynnvfrank committed:
FIX slide change events

lynnvfrank committed:
Re-enabled all events; Updated docs and example slideshow

lynnvfrank committed:
Updated the code examples to be more succinct

lynnvfrank committed:
Added a conclusion slide to the example

lynnvfrank committed:
FIX for menu not appearing because sections were not being assigned

lynnvfrank committed:
Fix for the goto command actions

lynnvfrank committed:
Moved the presentation creation methods closer to home

lynnvfrank committed:
FIX menu navigation left no longer breaks the menu

lynnvfrank committed:
VERSION 0.8.1

lynnvfrank committed:
Merge branch 'showoff-rename'

lynnvfrank committed:
Explicit blankslate version

lynnvfrank committed:
Removed debugger and updated source

lynnvfrank committed:
Initial replacement of pygments

lynnvfrank committed:
Adjusted CSS and JavaScript to work with Rouge element output

lynnvfrank committed:
Simplified and removed the .sh_ prefix for code execution

lynnvfrank committed:
Removed references to Pygments

lynnvfrank committed:
Added config.ru to example

lynnvfrank committed:
Additional backwards support for command-prompt examples

lynnvfrank committed:
Merge branch 'rouge-over-pygments'

lynnvfrank committed:
Version 0.9.0 - Rouge

lynnvfrank committed:
Removing the live-ruby code and adding an plugin interface

lynnvfrank committed:
Merge branch 'liveruby-plugin'

lynnvfrank committed:
Removed ruby execution from README

lynnvfrank committed:
Removed the live ruby code

lynnvfrank committed:
Clean up of the server

lynnvfrank committed:
Removed Preshow from the Ruby code

lynnvfrank committed:
Added plugin CSS support

lynnvfrank committed:
Moved preshow key command to plugin

lynnvfrank committed:
Removed CSS

lynnvfrank committed:
Removed remaining preshow js

lynnvfrank committed:
Adding the ability to add key commands to the help menu

lynnvfrank committed:
Merge branch 'preshow-plugin'

lynnvfrank committed:
Removed comment in HTML

lynnvfrank committed:
Removed the undefined constant for Preshow

lynnvfrank committed:
PDF presentation is now working

lynnvfrank committed:
Removing unused default.css

lynnvfrank committed:
Removed the PDF presentation path

lynnvfrank committed:
FIX: 'parade static pdf' command did not allow the output to be specified

lynnvfrank committed:
FIX: 'parade static html' command did not allow output to be specified

lynnvfrank committed:
Support for -o and --output command for static

lynnvfrank committed:
Merge branch 'pdf-path-removal'

lynnvfrank committed:
Theme Support

Parade files may now define a theme from the list of the following:

* archetect
* hack
* merlot
* slate

lynnvfrank committed:
Version 0.9.1

lynnvfrank committed:
Merge branch 'theme-support'

lynnvfrank committed:
Version 0.9.2

lynnvfrank committed:
Splitting out the help and the footer into their own templates

lynnvfrank committed:
Plugin css/js is using inline erb templates

thesteady committed:
fixed typo in bin/parade

ln 57 had "yourr"

lynnvfrank committed:
Merge pull request #44 from thesteady/patch-1

fixed typo in bin/parade

lynnvfrank committed:
Removed .full-page class

lynnvfrank committed:
Removed the full-page example and docs

gschorkopf committed:
Changed files to reflect a wider selection of font sizes

gschorkopf committed:
There was a typo on the text-size-percentage page

lynnvfrank committed:
Footer works on top level sections

thesteady committed:
initial fixing of typos

gschorkopf committed:
Removed the slide for code -monospaces and the section in the README

lynnvfrank committed:
README customizing your footer

lynnvfrank committed:
Merge pull request #46 from gschorkopf/smaller

Changed files to reflect a wider selection of font sizes

lynnvfrank committed:
Merge pull request #45 from burtlo/remove-full-page-style

Remove full page style

lynnvfrank committed:
Merge pull request #47 from burtlo/customizable-footer

Customizable footer

lynnvfrank committed:
Merge pull request #48 from gschorkopf/removes-code-slides

Removes code slides

lynnvfrank committed:
FIX for footer

lynnvfrank committed:
Removed .code styles

lynnvfrank committed:
Executable Ruby only with the plugin

lynnvfrank committed:
Slides can have the blank class

lynnvfrank committed:
FIX for integration specs based on example preso change

lynnvfrank committed:
Adding rspec option to not run integration by default

lynnvfrank committed:
Merge branch 'blank-slide'

lynnvfrank committed:
Slide Title CSS

lynnvfrank committed:
Updated the slide presentation to show off title and center

lynnvfrank committed:
Merge branch 'slide-title'

lynnvfrank committed:
Slide style title-and-content

lynnvfrank committed:
Merge branch 'slide-title_and_content'

lynnvfrank committed:
Slide section-header

lynnvfrank committed:
Merge branch 'slide-section_header'

lynnvfrank committed:
Slide class comparison

lynnvfrank committed:
Merge branch 'slide-comparison'

lynnvfrank committed:
CSS Clean Up

* Updated css and js helpers to allow multiple files
* Moved parade.css styles to themes/default.css
* Moved slide styles to the slide_formats.css
* Moved navigation menu things to their own partial template

lynnvfrank committed:
Merge branch 'css-cleanup'

lynnvfrank committed:
Refined the merlot theme to just the essentials

lynnvfrank committed:
Removed subsection slide style

lynnvfrank committed:
Refined the Archetect theme to essentials

lynnvfrank committed:
Minor spacing in merlot theme

lynnvfrank committed:
Refined hack theme; Added BSOD and ./

lynnvfrank committed:
Refined the slate theme

lynnvfrank committed:
Hammer.js for touch support

lynnvfrank committed:
Merge branch 'touch-events'

thesteady committed:
Merge branch 'master' of github.com:burtlo/parade into proofread_README

thesteady committed:
edits to the README

lynnvfrank committed:
Updated to support iPad portrait and landscape

lynnvfrank committed:
Merge branch 'mobile/medium-form-factor'

lynnvfrank committed:
Merge pull request #51 from thesteady/proofread_README

Proofread and edits to README

lynnvfrank committed:
iPhone landscape

lynnvfrank committed:
Merge branch 'mobile/small-form-factor'

lynnvfrank committed:
Setting Rouge '0.2.14' to workaround issues with loading on heroku

lynnvfrank committed:
iPad Portrait Format reduce the size of the font

lynnvfrank committed:
FIX: for the iPad landscape code size

lynnvfrank committed:
Better iPhone Landscape and Portrait Support

lynnvfrank committed:
FIX for #31 removed 'reload slides'

lynnvfrank committed:
Fixes for Print format and FIX for generation of static pdf

lynnvfrank committed:
Parade output in HTML and PDF are now correctly styled

* FIX an ugly border that was present
* Updated so that they support themes
* FIX extra page in-between each slide

lynnvfrank committed:
Merge branch 'print-format'

lynnvfrank committed:
Updated /print URL to commands and additional slide movement commands

lynnvfrank committed:
Cleaned up available commands in README Issue #34

lynnvfrank committed:
Adding Up and Down arrows to the help menu #33

lynnvfrank committed:
FIX: clean up of the print styles

lynnvfrank committed:
Renaming theme to jquery-ui

lynnvfrank committed:
FIX Issue #28 missing pause info from help menu

lynnvfrank committed:
Slide Image - Resizing

* Presentation will now load the slides through JS and not rely on
  the slides being in the div.
* FIX: /print will load the print.css and the normal view will not
* Images that are too large are resized to the size of the preso

lynnvfrank committed:
FIX for mobile formats of centered slides and print

lynnvfrank committed:
Merge branch 'js/slide-images-resolution'

lynnvfrank committed:
Version 0.10.0

lynnvfrank committed:
Moved presentation loading to index.erb out of parade.js

lynnvfrank committed:
Added Presentation Path Prefix

A presentation path prefix is set by the env['SCRIPT_NAME']. This
is restoring a feature that was added with the following pull
request:

https://github.com/schacon/showoff/pull/136/files

* All JS/CSS helpers will now recognize include the path prefix
* The JavaScript presentation will request the slides from the correct prefixed path
* The image paths are updated correctly as well

lynnvfrank committed:
Allow ImagePaths to be more tolerant of nils in prefixes

lynnvfrank committed:
Updated specs for UpdateImagePath to test presentation_path_prefix

lynnvfrank committed:
Merge branch 'multi-tenat-hosting'

lynnvfrank committed:
Version 0.10.1

lynnvfrank committed:
Cleaned up Future Plans

lynnvfrank committed:
Adding a Parade Banner

lynnvfrank committed:
Flipped Marcy in Parade Image

thesteady committed:
created a new css theme, hayfield

novohispano committed:
Minimal theme

lynnvfrank committed:
Example Preso - commented out the default.css overrides

lynnvfrank committed:
Merge remote-tracking branch 'laura/new_css_theme'

lynnvfrank committed:
Hayfield Theme - updated the footer style

lynnvfrank committed:
Example default.css really commented out all the styles this time

lynnvfrank committed:
Updated Theme list in README

lynnvfrank committed:
Merge remote-tracking branch 'jorge/working_on_css'

Conflicts:
	example/default.css

lynnvfrank committed:
Updated Minimal theme to have the correct background on command-line samples

lynnvfrank committed:
Adding Minimal Theme to the list of available themes

ivanoats committed:
Added example generate presentation command
because I got stuck thinking options were --thing=value

lynnvfrank committed:
Merge pull request #61 from SustainableWebsites/master

Added example generate presentation command

novohispano committed:
Adding bullet styles to minimal and minor changes on those slides in the tutorial

lalalainexd committed:
refactored columns renderer a lil bit

lalalainexd committed:
completed inital implementation for content-with-caption renderer

lalalainexd committed:
added example use of content with caption and fixed slide post renderer to user the correct renderer for content with caption

lynnvfrank committed:
Merge remote-tracking branch 'jorge/working_on_css'

lynnvfrank committed:
Merge remote-tracking branch 'elaine/master'

lynnvfrank committed:
Cleaned up spacing in the README

lynnvfrank committed:
Section Slide Classes

Allow the ability to define section specific slide styles and classes.

```
section "Iteration 0" do
  css_classes "greed-background"
  slides "iteration-zero.md"
end

section "Iteration 1" do
  css_classes "blue-background"
  slides "iteration-zero.md"
end
``

lynnvfrank committed:
Added a print command to the visor which takes the user to the print page

lynnvfrank committed:
FIX: Errors in print version when a slide contains JQuery

lynnvfrank committed:
FIX: README Notes Section #35

lynnvfrank committed:
Removed further mention of the presenter

lynnvfrank committed:
Re-arranged the documentation to place styles in a more general to specific order

lynnvfrank committed:
Updating the README to sell the project better

lynnvfrank committed:
Updated to add Octocat on Deck Failure

lynnvfrank committed:
Removing all support for the JSON file format

lynnvfrank committed:
Removing notes from ever being toggled

lynnvfrank committed:
Removing the images from the file list

lynnvfrank committed:
Adding the overriding slide docs

lynnvfrank committed:
Added the new section based css_classes

lynnvfrank committed:
Version 0.10.2
